sync/clients/sync.py

```diff
@@ -112,6 +112,21 @@ def get_project_cluster_template(self, project_id: str, params: dict = None) ->
             self._client.build_request("GET", f"/v1/projects/{project_id}/cluster/template", params=params)
         )
 
+    def get_project_clusters(
+        self,
+        workspace_id: str,
+        job_id: str,
+    ) -> dict:
```

Comment:
Can this return typing `dict` be updatesdto match the API response object `list[DatabricksJobCluster]`? 

 https://github.com/synccomputingcode/sync_backend/pull/945/files#diff-a1ddcccdbbd3130da76172fa9c228fc1e63f9a7de1b6fe2ad3374430cb1b4f1dR892